cpp/src/utilities/cython.cu

```diff
@@ -476,6 +476,185 @@ std::pair<size_t, weight_t> call_louvain(raft::handle_t const& handle,
     handle, graph_container, functor);
 }
 
+// Wrapper for calling Pagerank through a graph container
+template <typename vertex_t, typename weight_t>
+void call_pagerank(raft::handle_t const& handle,
+                   graph_container_t const& graph_container,
+                   vertex_t* identifiers,
+                   weight_t* p_pagerank,
+                   vertex_t personalization_subset_size,
+                   vertex_t* personalization_subset,
+                   weight_t* personalization_values,
+                   double alpha,
+                   double tolerance,
+                   int64_t max_iter,
+                   bool has_guess)
+{
+  if (graph_container.graph_type == graphTypeEnum::GraphCSCViewFloat) {
+    pagerank(handle,
+             *(graph_container.graph_ptr_union.GraphCSCViewFloatPtr),
+             reinterpret_cast<float*>(p_pagerank),
+             static_cast<int32_t>(personalization_subset_size),
+             reinterpret_cast<int32_t*>(personalization_subset),
+             reinterpret_cast<float*>(personalization_values),
+             alpha,
+             tolerance,
+             max_iter,
+             has_guess);
+    graph_container.graph_ptr_union.GraphCSCViewFloatPtr->get_vertex_identifiers(
+      reinterpret_cast<int32_t*>(identifiers));
+  } else if (graph_container.graph_type == graphTypeEnum::GraphCSCViewDouble) {
+    pagerank(handle,
+             *(graph_container.graph_ptr_union.GraphCSCViewDoublePtr),
+             reinterpret_cast<double*>(p_pagerank),
+             static_cast<int32_t>(personalization_subset_size),
+             reinterpret_cast<int32_t*>(personalization_subset),
+             reinterpret_cast<double*>(personalization_values),
+             alpha,
+             tolerance,
+             max_iter,
+             has_guess);
+    graph_container.graph_ptr_union.GraphCSCViewDoublePtr->get_vertex_identifiers(
+      reinterpret_cast<int32_t*>(identifiers));
+  } else if (graph_container.graph_type == graphTypeEnum::graph_t) {
+    if (graph_container.edgeType == numberTypeEnum::int32Type) {
```

Comment:
This would be cleaner (for pagerank, BFS and SSSP) to use the technique used by louvain...  define a functor for each algorithm and then call `detail::call_function<true, void>(handle, graph_container, functor);`

Reply:
Although we could defer this work to 0.17